src/Units/AWBWUnits.java

```diff
@@ -0,0 +1,778 @@
+import Units.MoveTypes.FloatHeavy;
+import Units.MoveTypes.FloatLight;
+import Units.MoveTypes.FootMech;
+import Units.MoveTypes.FootStandard;
+import Units.MoveTypes.MoveType;
+import Units.MoveTypes.Tires;
+import Units.MoveTypes.Tread;
+
+public class AWBWUnits extends UnitModelScheme
+{
+  private static final long serialVersionUID = 1L;
+
+  @Override
+  public String toString()
+  {
+    return "AWBW" + (schemeValid ? "" : "!");
+  }
+
+  @Override
+  public String getIconicUnitName()
+  {
+    return "Infantry";
+  }
+
+  @Override
+  public GameReadyModels getGameReadyModels()
+  {
+    GameReadyModels awbwModels = new GameReadyModels();
+
+    ArrayList<UnitModel> factoryModels = new ArrayList<UnitModel>();
+    ArrayList<UnitModel> seaportModels = new ArrayList<UnitModel>();
+    ArrayList<UnitModel> airportModels = new ArrayList<UnitModel>();
+
+    // Define everything we can build from a Factory.
+    factoryModels.add(new InfantryModel());
+    factoryModels.add(new MechModel());
+    factoryModels.add(new APCModel());
+    factoryModels.add(new ArtilleryModel());
+    factoryModels.add(new ReconModel());
+    factoryModels.add(new TankModel());
+    factoryModels.add(new MDTankModel());
+    factoryModels.add(new NeotankModel());
+    factoryModels.add(new MegatankModel());
+    factoryModels.add(new RocketsModel());
+    factoryModels.add(new AntiAirModel());
+    factoryModels.add(new MobileSAMModel());
+    factoryModels.add(new PiperunnerModel());
+
+    // Record those units we can get from a Seaport.
+    seaportModels.add(new LanderModel());
+    seaportModels.add(new CruiserModel());
+    UnitModel sub = new SubModel();
+    seaportModels.add(sub);
+    seaportModels.add(new BattleshipModel());
+    seaportModels.add(new CarrierModel());
+    seaportModels.add(new BBoatModel());
+
+    // Inscribe those war machines obtainable from an Airport.
+    airportModels.add(new TCopterModel());
+    airportModels.add(new BCopterModel());
+    airportModels.add(new FighterModel());
+    airportModels.add(new BomberModel());
+    UnitModel stealth = new StealthModel();
+    airportModels.add(stealth);
+    airportModels.add(new BBombModel());
+
+    // Dump these lists into a hashmap for easy reference later.
+    awbwModels.shoppingList.put(TerrainType.FACTORY, factoryModels);
+    awbwModels.shoppingList.put(TerrainType.SEAPORT, seaportModels);
+    awbwModels.shoppingList.put(TerrainType.AIRPORT, airportModels);
+
+    // Compile one master list of everything we can build.
+    for (UnitModel um : factoryModels)
+      awbwModels.unitModels.add(um);
+    for (UnitModel um : seaportModels)
+      awbwModels.unitModels.add(um);
+    for (UnitModel um : airportModels)
+      awbwModels.unitModels.add(um);
+
+    // Handle transforming units separately, since we don't want two buy-entries
+    UnitModel subsub = new SubSubModel();
+    sub.possibleActions.add(new UnitActionType.Transform(subsub, "DIVE"));
```

Comment:
Why are you adding unit-specific actions here instead of in the `UnitModel` class itself?

Reply:
Because it requires another `UnitModel` to create the action, since it's a transformation.
I suppose I could build `sub` inside `subsub`, but there's no interface currently for generating required *related* `UnitModel`s.